go-controller/pkg/node/gateway.go

```diff
@@ -408,51 +416,110 @@ func (b *bridgeConfiguration) updateInterfaceIPAddresses() ([]*net.IPNet, error)
+		bypassRep:  bypassRep,
+		uplinkName: uplink,
+		macAddress: macAddress,
+	}, nil
+}
+
+func bridgeForRegularInterface(iface string) (*bridgeConfiguration, string, error) {
+	var bridgeName, uplinkName string
+	var err error
+	gwIntf := iface
+
+	if bridgeName, _, err = util.RunOVSVsctl("port-to-br", iface); err == nil {
+		// This is port of an OVS bridge
+		if uplinkName, err = util.GetNicName(bridgeName); err != nil {
+			return nil, "", errors.Wrapf(err, "Failed to find nic name for bridge %s", bridgeName)
+		}
+	} else if _, _, err = util.RunOVSVsctl("br-exists", iface); err != nil {
+		// This is not an OVS bridge or bridge doesn't exist. We need to create it
 		// and add cluster.GatewayIntf as a port of that bridge.
-		bridgeName, err := util.NicToBridge(intfName)
-		if err != nil {
-			return nil, errors.Wrapf(err, "NicToBridge failed for %s", intfName)
+		if bridgeName, err = util.NicToBridge(iface); err != nil {
+			return nil, "", errors.Wrapf(err, "NicToBridge failed for %s", iface)
 		}
-		res.bridgeName = bridgeName
-		res.uplinkName = intfName
+		uplinkName = iface
 		gwIntf = bridgeName
-	} else {
+	} else if uplinkName, err = getIntfName(iface); err == nil {
 		// gateway interface is an OVS bridge
-		uplinkName, err := getIntfName(intfName)
-		if err != nil {
-			return nil, errors.Wrapf(err, "Failed to find intfName for %s", intfName)
-		}
-		res.bridgeName = intfName
-		res.uplinkName = uplinkName
+		bridgeName = iface
+	} else {
+		return nil, "", errors.Wrapf(err, "Failed to find uplink name for %s", iface)
+	}
+
+	macAddress, err := util.GetOVSPortMACAddress(gwIntf)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "failed to get MAC address for ovs port %s", gwIntf)
 	}
+
+	return &bridgeConfiguration{
+		bridgeName: bridgeName,
+		uplinkName: uplinkName,
+		macAddress: macAddress,
+	}, gwIntf, nil
+}
+
+func getRepresentorAndUplinkName(intfName string) (string, string, error) {
+	deviceID, err := util.GetDeviceIDFromNetdevice(intfName)
+	if err != nil {
+		return "", "", err
+	}
+
+	return util.GetFunctionRepresentorName(deviceID)
+}
+
+func bridgeForInterface(intfName, nodeName, physicalNetworkName string, kubeNodeIP net.IP) (*bridgeConfiguration, error) {
+	var res *bridgeConfiguration
+	var gwIntf, rep, uplink string
 	var err error
-	// Now, we get IP addresses for the bridge
+
+	// Try to get representor and uplink name for the specified device.
+	// If function had succeed, then it is either switchdev VF or SF, hence, configure it as a bypass interface.
+	// If failed - falback to regular interface configuration.
+	rep, uplink, err = getRepresentorAndUplinkName(intfName)
+	if err == nil {
+		gwIntf = intfName
+		if res, err = bridgeForBypassInterface(uplink, intfName, rep); err != nil {
+			return nil, errors.Wrapf(err, "Failed to configure bridge layout for %s", intfName)
+		}
+	} else {
+		klog.Warningf("%s not a bypass interface (err - %v), falling back to regular interface configuration", intfName, err)
```

Comment:
Maybe warning log level is a bit too much for the "normal" case.

Reply:
Right.